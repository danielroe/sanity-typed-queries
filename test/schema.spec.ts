import {
  defineDocument,
  defineFields,
  defineObject,
} from 'sanity-typed-queries'
import { describe, expect, it } from 'vitest'

describe('schema creator', () => {
  it('generates correct schema fields for document', () => {
    const { document } = defineDocument('typeOfDocument', {
      name: {
        type: 'string',
        validation: Rule => Rule.required(),
      },
      postcode: {
        type: 'string',
        validation: Rule => Rule.required(),
      },
      cost: {
        type: 'number',
      },
      picture: {
        type: 'image',
      },
      description: {
        type: 'text',
        rows: 2,
        validation: Rule => Rule.required(),
      },
    })

    expect(document).toMatchSnapshot()
  })

  it('generates correct schema fields for object', () => {
    const { object } = defineObject('myObject', {
      name: {
        type: 'string',
        validation: Rule => Rule.required(),
      },
    })

    expect(object).toMatchSnapshot()
  })

  it('generates correct schema fields when fields are defined', () => {
    expect(defineFields({ field1: { type: 'string' } })).toMatchSnapshot()
  })
})
