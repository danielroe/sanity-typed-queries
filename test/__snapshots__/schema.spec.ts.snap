// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`schema creator > generates correct schema fields for document 1`] = `
{
  "fields": [
    {
      "name": "name",
      "title": "Name",
      "type": "string",
      "validation": [Function],
    },
    {
      "name": "postcode",
      "title": "Postcode",
      "type": "string",
      "validation": [Function],
    },
    {
      "name": "cost",
      "title": "Cost",
      "type": "number",
    },
    {
      "name": "picture",
      "title": "Picture",
      "type": "image",
    },
    {
      "name": "description",
      "rows": 2,
      "title": "Description",
      "type": "text",
      "validation": [Function],
    },
  ],
  "name": "typeOfDocument",
  "title": "Type Of Document",
  "type": "document",
}
`;

exports[`schema creator > generates correct schema fields for object 1`] = `
{
  "fields": [
    {
      "name": "name",
      "title": "Name",
      "type": "string",
      "validation": [Function],
    },
  ],
  "name": "myObject",
  "title": "My Object",
  "type": "object",
}
`;

exports[`schema creator > generates correct schema fields when fields are defined 1`] = `
[
  {
    "name": "field1",
    "title": "Field1",
    "type": "string",
  },
]
`;
